global-privacy-control/tests.json

```diff
@@ -0,0 +1,129 @@
+{
+    "gpcHeader": {
+        "name": "Tests related to the GPC request header",
+        "desc": "Check if `Sec-GPC: 1` is correctly sent with all requests",
+        "tests": [
+            {
+                "name": "Header injected - navigational requests",
+                "siteURL": "http://global-privacy-control.glitch.me",
+                "requestURL": "http://global-privacy-control.glitch.me",
+                "requestType": "main_frame",
+                "gpcUserSettingOn": true,
+                "expectGPCHeader": true,
+                "expectGPCHeaderValue": "1",
+                "exceptPlatforms": []
+            },
+            {
+                "name": "Header not injected - user opted out of GPC",
+                "siteURL": "http://global-privacy-control.glitch.me",
+                "requestURL": "http://global-privacy-control.glitch.me",
+                "requestType": "main_frame",
+                "gpcUserSettingOn": false,
+                "expectGPCHeader": false,
+                "exceptPlatforms": []
+            },
+            {
+                "name": "Header not injected - privacy config local exception (gpc/exceptions)",
+                "siteURL": "https://www.washingtonpost.com/",
+                "requestURL": "https://www.washingtonpost.com/",
+                "requestType": "main_frame",
+                "gpcUserSettingOn": true,
+                "expectGPCHeader": false,
+                "exceptPlatforms": []
+            },
+            {
+                "name": "Header not injected - privacy config global exception (unprotectedTemporary)",
+                "siteURL": "https://globalprivacycontrol.org/",
+                "requestURL": "https://globalprivacycontrol.org/",
+                "requestType": "main_frame",
+                "gpcUserSettingOn": true,
+                "expectGPCHeader": false,
+                "exceptPlatforms": []
+            },
+            {
+                "name": "Header injected - subrequest",
+                "siteURL": "https://example.com/",
+                "requestURL": "https://example.com/image.png",
+                "requestType": "image",
+                "gpcUserSettingOn": true,
+                "expectGPCHeader": true,
+                "exceptPlatforms": [
+                    "ios-browser",
+                    "macos-browser",
+                    "android-browser"
+                ]
+            },
+            {
+                "name": "Header not injected - subrequest on a excluded site (gpc/exceptions)",
+                "siteURL": "https://www.washingtonpost.com/",
```

Comment:
It would be great to move away from using production commercial domains in tests and use dedicated test sites/values.

Reply:
Agreed, I only used those real domains because our apps limit this feature to just a couple hard coded domains, so this is an exception.